### PR TITLE
[clang] Add bitint classification for __builtin_classify_type

### PR DESCRIPTION
**Pull request by tbaederr**
See #71911

### CODE REVIEW

File: llvm/cmake/modules/HandleLLVMOptions.cmake [hunk 1]
```diff
@@ -1060,7 +1060,7 @@ if (UNIX AND
     (CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR
      (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
       NOT (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.9))))
-  append("-fdiagnostics-color" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+  append("-fdiagnostics-color=always" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
 endif()
```
Comment (cor3ntin) [inline llvm/cmake/modules/HandleLLVMOptions.cmake:1064]: This is unrelated

**Review by cor3ntin — COMMENTED** [review]
Can you add a release note? otherwise this looks good

File: llvm/cmake/modules/HandleLLVMOptions.cmake (hunk 1 above)
Comment (tbaederr) [reply llvm/cmake/modules/HandleLLVMOptions.cmake]: Didn't even work :/

File: clang/lib/AST/ExprConstant.cpp [hunk 2]
```diff
@@ -11486,6 +11486,7 @@ bool IntExprEvaluator::CheckReferencedDecl(const Expr* E, const Decl* D) {
 
 /// Values returned by __builtin_classify_type, chosen to match the values
 /// produced by GCC's builtin.
+/// The values can be found in gcc/typeclass.h in the GCC repository.
```
Comment (AaronBallman) [inline clang/lib/AST/ExprConstant.cpp:11489]:
```suggestion

```
For licensing clarity, I don't think we want to encourage people to go look at the source of GCC even though this is a helpful comment.

**Review by AaronBallman — APPROVED** [review]
LGTM

File: clang/docs/ReleaseNotes.rst [hunk 3]
```diff
@@ -217,6 +217,8 @@ Non-comprehensive list of changes in this release
   (e.g., ``uint16x8_t``), this returns the constant number of elements at compile-time.
   For scalable vectors, e.g., SVE or RISC-V V, the number of elements is not known at compile-time and is
   determined at runtime.
+* ``__builtin_classify_type()`` now classifies ``_BitInt`` values as the return value ``18``,
+  to match GCC 14's behavior.
```
Comment (AaronBallman) [inline clang/docs/ReleaseNotes.rst:221]: We should mention that this fixes https://github.com/llvm/llvm-project/issues/71911

Comment (tbaederr) [reply clang/docs/ReleaseNotes.rst:221]: I didn't add that because the issue also  mentions some extensions to `__builtin_classify_type`. We should probably split that out into a separate issue where we also document the exact extensions, maybe with a link to the gcc commit and test cases.

Comment (AaronBallman) [reply clang/docs/ReleaseNotes.rst:223]: Okay, that's reasonable.

**Review by AaronBallman — APPROVED** [review]
